### PR TITLE
#155931587 Implement API Access

### PR DESCRIPTION
**Pull request by NabunyaLilian**
**What does this PR do**
- Implements access to the Github API 

**Tasks to be completed**
- create a presenter file
- create a view for github users
- Fetch users using the retrofit instance

**Relevant PT stories**
[#155931587](https://www.pivotaltracker.com/story/show/155931587)

### CODE REVIEW

File: app/src/main/java/com/example/thecodelab/presenter/GithubPresenter.java [hunk 1]
```diff
@@ -1,4 +1,52 @@
...
+
+                        if (githubUsersResponse != null && githubUsersResponse.getUsers() !=null){
+                            List<GithubUser> githubUser = githubUsersResponse.getUsers();
+                            githubUsersView.readyUsers(githubUser);
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Call<GithubUsersResponse> call, Throwable t) {
+                        Log.d("TAG", "onFailure: returned: " + t);
```
Comment (joshNic) [inline app/src/main/java/com/example/thecodelab/presenter/GithubPresenter.java]: @NabunyaLilian  could you please remove the logs (these should only be used to debug and should not be pushed into production) you could look into retrofit interceptors for an efficient implementation of this.

File: app/src/main/java/com/example/thecodelab/service/GithubApi.java [hunk 2]
```diff
@@ -7,6 +7,7 @@
 import retrofit2.http.GET;
 import retrofit2.http.Path;
 
+
 public interface GithubApi {
 
     @GET("search/users?q=type:User+location:Nairobi+language:JAVA")
```
Comment (joshNic) [inline app/src/main/java/com/example/thecodelab/service/GithubApi.java:13]: @NabunyaLilian there is a better way of doing this using retrofit's @QueryMap annotation. [(Link)](https://square.github.io/retrofit/2.x/retrofit/index.html?retrofit2/http/QueryMap.html)

**Comment by NabunyaLilian** [comment]
@joshNic, thanks the feedback let me work on it.

**Review by joshNic — APPROVED** [review]
LGTM am merging this

**Pull request merged**